### PR TITLE
feat: allow ticks to be configurable with total count, interval, and starting placement

### LINKED ISSUE #79: Can't control tick generation
It's desirable to format ticks using a [binary unit prefix](https://en.wikipedia.org/wiki/Binary_prefix) when plotting data whose y-value is measured in bytes. 

But the y-axis generates tick marks at "nice" intervals, i.e. at powers of 10. These intervals are no longer nice when formatting numbers with a binary unit prefix.

For example, here's the same plot with ticks formatted using a decimal unit prefix:

Screen Shot 2019-06-04 at 2 58 24 PM

It has a nice tick of "6G". When formatting the ticks with a binary unit prefix, it looks like this:

Screen Shot 2019-06-04 at 2 58 31 PM

The generated tick is now "5.5879G" (6 gigabytes =~ 5.88 gibibytes), which is no longer nice.

### PR DESCRIPTION
**Pull request by TCL735**
Closes #79 

Allows user to specify tick generation in the config.

Screen Shot 2020-10-21 at 8 52 15 PM

### CODE REVIEW

File: giraffe/src/utils/getTextMetrics.ts [hunk 1]
```diff
@@ -18,9 +20,18 @@ export const getTextMetrics = (font: string, text: string): TextMetrics => {
...
 
   span.innerText = text
-
+  spanWithDash.innerText = '-'
+
+  // Text with the same number of characters:
+  //   when it includes a dash (negative number), it tends to be skinnier than
+  //   a positive number because a dash is not as wide as a single digit (most of the time).
+  //   Add padding when text has a dash by making dashes twice as wide.
+  const widthPadding =
```
Comment (hoorayimhelping) [inline giraffe/src/utils/getTextMetrics.ts:29]: is there a cleaner way of doing this? i don't think we can rely on a dash being the same thing as a negative indicator. some platforms might not render `-1` consistently with other platforms

File: giraffe/src/utils/getTicks.test.ts [hunk 2]
```diff
@@ -1,13 +1,293 @@
...
+          valueDomain,
+          VALUE,
+          totalTicks,
+          tickStart,
+          tickStep
+        )
+        expect(result.length).toEqual(0)
+      })
+    })
+  })
```
Comment (hoorayimhelping) [inline giraffe/src/utils/getTicks.test.ts:288]: great tests from a tactical level. from a higher, strategic level, just reinforcing what i said before - there's a lot of interplay here that is pretty complex and will probably be really confusing for users and might be an indication of larger design issues.

File: giraffe/src/utils/getTicks.ts [hunk 3]
```diff
@@ -112,20 +114,93 @@ export const getTicks = (
...
+  domain: number[],
+  rangeLength: number,
+  tickSize: number,
+  columnKey: string,
+  totalTicks?: number,
+  tickStart?: number,
+  tickStep?: number
+) => {
+  const [start = 0, end = 0] = domain
+  if (isFiniteNumber(totalTicks) || isFiniteNumber(tickStep)) {
```
Comment (hoorayimhelping) [inline giraffe/src/utils/getTicks.ts:175]: should this `||` be an `&&`?

File: giraffe/src/utils/isFiniteNumber.test.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    expect(isFiniteNumber(undefined)).toEqual(false)
+    expect(isFiniteNumber(null)).toEqual(false)
+    expect(isFiniteNumber({})).toEqual(false)
+    expect(isFiniteNumber([])).toEqual(false)
+    expect(isFiniteNumber('123')).toEqual(false)
+    expect(isFiniteNumber('-0')).toEqual(false)
+    expect(isFiniteNumber(() => 123)).toEqual(false)
+    expect(isFiniteNumber(Symbol())).toEqual(false)
+  })
+})
```
Comment (hoorayimhelping) [inline giraffe/src/utils/isFiniteNumber.test.ts:45]: for consistency, can you change these `test`s to `it`s please?

**Review by hoorayimhelping — APPROVED** [review]
The code looks fine in isolation, but the higher level concerns about the explosion of configuration options still remain for me

File: giraffe/src/utils/getTextMetrics.ts (hunk 1 above)
Comment (TCL735) [reply giraffe/src/utils/getTextMetrics.ts:29]: Yes there is a cleaner way. No extra `<span>` necessary. We do need to check for a dash though, unfortunately. I will send up a new commit in the near future.

File: giraffe/src/utils/getTicks.ts (hunk 3 above)
Comment (TCL735) [reply giraffe/src/utils/getTicks.ts:175]: No. It is correct as `||`. I will demo this to show off the flexibility.

Comment (TCL735) [reply giraffe/src/utils/getTicks.ts:175]: Sure, will do.

File: giraffe/src/utils/isFiniteNumber.test.ts (hunk 4 above)
Comment (TCL735) [reply giraffe/src/utils/isFiniteNumber.test.ts:45]: Sure, will do.

File: giraffe/README.md [hunk 5]
```diff
@@ -258,9 +258,21 @@ When using the comma separated values (CSV) from the Flux query as the `fluxResp
...
+
+- **yTotalTicks**: _number. Optional. **Ignored when yTicks is specified**._ A number representing the maximum possible number of ticks to generate on the y-axis. Uses the **yTickStep** as the tick interval if also included. Otherwise the tick interval is taken from dividing the length of the rendered domain by this number. The actual number of rendered ticks may be less than this number due to the size of the tick interval.
+
+- **yTickStep**: _number. Optional. **Ignored when yTicks is specified**._ A number representing the tick interval for the y-axis. May be negative.
+
+- **yTickStart**: _number. Optional. **Ignored when yTicks is specified**._ A number representing the value of the first tick on the y-axis. This will determine the placement of all subsequent ticks. Must be within the domain of the y-axis, otherwise will not be rendered. Subsequent ticks may still be rendered if their placement is within the domain.
```
[the remark this answers is not in the document]
Comment (TCL735) [reply giraffe/README.md:276]: Slight clarification and change to the text of `xTickStart` and `yTickStart`.

What do you think of this?
Screen Shot 2020-10-22 at 3 57 30 PM

**Pull request merged**